ocaml/xapi/xapi_xenops.ml

```diff
@@ -4221,21 +4225,25 @@ let vif_unplug ~__context ~self force =
       info "xenops: VIF.unplug %s.%s" (fst vif.Vif.id) (snd vif.Vif.id) ;
       let dbg = Context.string_of_task __context in
       let module Client = (val make_client queue_name : XENOPS) in
-      Client.VIF.unplug dbg vif.Vif.id force
-      |> sync_with_task __context queue_name ;
-      (* We need to make sure VIF.stat still works so: wait before calling VIF.remove *)
-      Events_from_xenopsd.wait queue_name dbg (fst vif.Vif.id) () ;
-      if Db.VIF.get_currently_attached ~__context ~self then
-        raise
-          Api_errors.(
-            Server_error
-              ( internal_error
-              , [
-                  Printf.sprintf "vif_unplug: Unable to unplug VIF %s"
-                    (Ref.string_of self)
-                ]
-              )
-          )
+      try
+        Client.VIF.unplug dbg vif.Vif.id force
+        |> sync_with_task __context queue_name ;
+        (* We need to make sure VIF.stat still works so: wait before calling VIF.remove *)
+        Events_from_xenopsd.wait queue_name dbg (fst vif.Vif.id) () ;
+        if Db.VIF.get_currently_attached ~__context ~self then
+          raise
+            Api_errors.(
+              Server_error
+                ( internal_error
+                , [
+                    Printf.sprintf "vif_unplug: Unable to unplug VIF %s"
```

Comment:
This can also use `__FUNCTION__` to help finding the origin of the message.

Reply:
That may not be appropriate in an API error; I think it's more useful for logging. I could add it to the new log line below, but if we thing this is useful, shouldn't we add it to the log library itself rather than having to manually add it to every log line?